### PR TITLE
Use OpenXLA theming on website.

### PR DESCRIPTION
**Pull request by ScottTodd**
As has been announced, we'll be moving IREE into the OpenXLA project and GitHub organization (https://github.com/iree-org/iree/issues/12102). This starts to update our documentation (https://iree-org.github.io/iree/ for now, this will change) to more closely match the style of other OpenXLA projects.

As more of the broader OpenXLA project is developed, we'll be tweaking the website to tell a consistent story.
* IREE will remain a modular underlying technology
* Sections like the blog and user-journey documentation may move to other OpenXLA subdomain websites
* Sections like API references may move in (we've had https://github.com/iree-org/iree/issues/5477 sitting for a while)

---

Demo for review: https://scotttodd.github.io/iree/

Most of these changes are adapted from the draft at https://github.com/iree-org/iree/compare/a420b8d179d39d9741893dd1a7792e75c1e9b083...scottamain:iree:site ([demo here]( https://scottamain.github.io/iree/)).

---

## Accepted changes:

* OpenXLA colors for light mode and dark mode
* Warning -> Caution, per https://developers.google.com/style/notices#pick-a-notice-type
* Use the right "permalink" icon
* Style tweaks to `mkdocs.yml`
* New font (Noto: https://fonts.google.com/noto)
* OpenXLA logo and favicon, with tweaks to the homepage button in the header and nav

## Not yet implemented changes:

* `stackoverflow-[light,dark].css` themes for highlightjs (I'm not sure how these should be set up and if we want them yet)
* Navigation and page restructuring/reorganization. Some restructuring is needed, but I'm tempted to hold off until more of OpenXLA settles (we might be able to point elsewhere for "getting started with JAX", for example)

## Rejected changes:

* "The IREE Authors" -> "The OpenXLA Authors" (IREE has its own `AUTHORS` file, copyright header, etc.)
* Use the same icon for "dark mode -> light mode" and "light mode -> dark mode" (I like the icon showing what the current setting is)

skip-ci: documentation and supporting infrastructure changes only

### CODE REVIEW

**Comment by benvanik** [comment]
my 2c: since IREE isn't openxla but a part of it I think having the two logos that link back to each respective homepage makes sense - same with favicons - when you have 30 tabs open being able to quickly identify what's what is really useful
(that is, if we have the same favicon but different content on our tld that's confusing, as would be clicking the openxla logo but going to the iree site - and they'll always be split as we aren't going to host our own project docs in another repo)

**Comment by ScottTodd** [comment]
> my 2c: since IREE isn't openxla but a part of it I think having the two logos that link back to each respective homepage makes sense - same with favicons - when you have 30 tabs open being able to quickly identify what's what is really useful (that is, if we have the same favicon but different content on our tld that's confusing, as would be clicking the openxla logo but going to the iree site - and they'll always be split as we aren't going to host our own project docs in another repo)

Agreed. Here's [a page on the draft site](https://scottamain.github.io/iree/blog/2021-10-15-cuda-backend/#flatbuffer-definition): 
image

* The header doesn't say "IREE" at all or have a link to IREE's home
* The tab title _does_ have `- IREE`
* The favicon is OpenXLA's

And here's [the same page on the site today](https://iree-org.github.io/iree/blog/2021-10-15-cuda-backend/#flatbuffer-definition):
image

So I think we want to keep the ghost favicon and find some way to get the two logos / homepage links onto the header.

**Comment by benvanik** [comment]
yeah, that'd be great - having consistent branding is good but not at the cost of usability :)

**Comment by ScottTodd** [comment]
Had some offline discussions about the favicon / logo. Going to standardize on the OpenXLA favicon and logo, but I did add a second header link to IREE's homepage. We're also tweaking some of the favicon details and considering variants for each project.

File: third_party/mkdocs-material/overrides/partials/header.html [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      title="IREE"
+      class="md-header__button md-logo"
+      aria-label="IREE"
+      data-md-component="logo"
+    >
+      <div class="md-header__title">
+          <div class="md-nav__title">
+            {{ config.site_name }}
+          </div>
+      </div>
```
Comment (ScottTodd) [inline third_party/mkdocs-material/overrides/partials/header.html]: Need to set some class on this that respects the color scheme
image
image

(also why is the nav indented...? might need to tweak the `<div>` I added there...)

File: docs/website/mkdocs.yml [hunk 2]
```diff
@@ -1,59 +1,61 @@
 site_name: IREE
 site_url: https://iree-org.github.io/iree/
+repo_url: https://github.com/iree-org/iree
+repo_name: iree-org/iree
+edit_uri: "" # Omit edit button (linking to GitHub) from each document
```
Comment (GMNGeoffrey) [inline docs/website/mkdocs.yml]: Looks like this was just moved, so keep as is in this PR, but I quite like having links to edit stuff on documentation. I frequently try to fix errors in documentation and get frustrated when it's not clear how to do that.

File: docs/website/mkdocs.yml [hunk 3]
```diff
@@ -1,59 +1,61 @@
...
 
-repo_url: https://github.com/iree-org/iree
-repo_name: iree-org/iree
-edit_uri: "" # Omit edit button (linking to GitHub) from each document
-
 extra:
+  # TODO(scotttodd): https://openxla.github.io/community (when hosted)
+  community_homepage: https://github.com/openxla/community
+
+  # TODO(scotttodd): join mkdocs-material insiders and remove notice?
```
Comment (GMNGeoffrey) [inline docs/website/mkdocs.yml:57]: What notice?

File: third_party/mkdocs-material/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  custom_dir: overrides
+  # Then symlink to files from overrides/partials/... to this directory
+```
+
+or
+
+```yaml
+  custom_dir: ../../third_party/mkdocs-material/overrides
+  # Then have no override files outside of this third_party/ directory
+```
```
Comment (GMNGeoffrey) [inline third_party/mkdocs-material/README.md]: I would just say the one that we're doing (symlinking). I'd also mention that they're in third_party and we're doing the symlinking to make the third-party code distinction clear

File: docs/website/mkdocs.yml (hunk 2 above)
Comment (ScottTodd) [reply docs/website/mkdocs.yml]: https://www.mkdocs.org/user-guide/configuration/#edit_uri I tried tweaking and removing this but I don't see the edit button... maybe it got removed or changed to opt-in at some point.

File: docs/website/mkdocs.yml (hunk 3 above)
Comment (ScottTodd) [reply docs/website/mkdocs.yml:57]: The notice in the footer that says which site generator was used: 
image

https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-footer/?h=#generator-notice

File: third_party/mkdocs-material/overrides/partials/header.html [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+            {% else %}
+              {{ page.title }}
+            {% endif %}
+          </span>
+        </div>
+      </div>
+    </div>
+
+    <!-- Color palette -->
+    {% if not config.theme.palette is mapping %}
```
Comment (pcoet) [inline third_party/mkdocs-material/overrides/partials/header.html:81]: Not sure what is happening here, with the `is mapping`. Please make sure it's doing the right thing. :-)

**Review by pcoet — APPROVED** [review]

File: third_party/mkdocs-material/overrides/partials/header.html (hunk 5 above)
Comment (ScottTodd) [reply third_party/mkdocs-material/overrides/partials/header.html:81]: That's part of the unchanged original header.html partial from https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/header.html

I forked the partials (into third_party/mkdocs-material) then made edits, which you can see in the commit history: https://github.com/iree-org/iree/pull/12229/commits/07f0a575d259fd46583ca86a910f2daff7c13a65#diff-fbe2def9f6a4c68a91e91b5903d10fb8173120a3450e8fd05c1d27ee9ca486ef

Comment (ScottTodd) [reply third_party/mkdocs-material/overrides/partials/header.html:81]: (That's the recommended way to extend a theme - see https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme, which I've also linked to from third_party/mkdocs-material/README.md)

File: docs/website/docs/assets/stylesheets/openxla.css [hunk 6]
```diff
@@ -0,0 +1,89 @@
+@import url(//fonts.googleapis.com/icon?family=Material+Icons|Material+Icons+Outlined);
```
Comment (GMNGeoffrey) [inline docs/website/docs/assets/stylesheets/openxla.css:1]: Is this file the source of truth for the styling? Is there another copy elsewhere. Would be good to document where this came from. If it's the only version checked-in right now, that's fine, but eventually it seems like IREE shouldn't be the source of truth for this file and we should have a link to the canonical version

File: docs/website/overrides/404.html [hunk 7]
```diff
@@ -1,9 +1,7 @@
 {% extends "main.html" %}
 {% block content %}
 <h1>
-  <img alt="👻" class="twemoji" src="https://iree-org.github.io/iree/ghost.svg" title=":ghost:">
```
Comment (GMNGeoffrey) [inline docs/website/overrides/404.html:4]: :'-(

**Review by GMNGeoffrey — APPROVED** [review]

File: docs/website/docs/assets/stylesheets/openxla.css (hunk 6 above)
Comment (ScottTodd) [reply docs/website/docs/assets/stylesheets/openxla.css:1]: TBD. For now I'm just copying the file that was added in the draft (after tweaking a few colors to match the latest spec). Once we have a few repos with published sites we will indeed want to share somehow.

File: docs/website/overrides/404.html (hunk 7 above)
Comment (ScottTodd) [reply docs/website/overrides/404.html:4]: Mostly an artifact of moving the ghost icon around and not wanting to keep multiple copies of the same file in the repo. It's no longer used for the favicon, but it is used for a custom emoji. The custom emoji implementation does some magic that's different from a regular image, and the 404 page is html that bypasses that magic during site generation.

(TL;DR: could add the ghost emoji back to the 404 page, but would need to add back another copy of the .svg file or see if symlinks work for that too)

**Pull request merged**